### DISCUSSION TITLE
php artisan migrate error no such file or directory

### DISCUSSION THREAD

**Discussion by Behinder** [laravel/framework#49886] [Q&A]
### Laravel Version

latest

### PHP Version

8.2.14

### Database Driver & Version

mysql  Ver 8.0.35 for macos13.5 on arm64 (Source distribution)

### Description

Tryint to run `php artisan migrate` on fresh install I got
````
SQLSTATE[HY000] [2002] No such file or directory (Connection: mysql, SQL: select table_name as `name`, (data_length + index_length) as `size`, table_comment as `comment`, engine as `engine`, table_collation as `collation` from information_schema.tables where table_schema = 'nocuberniego' and table_type in ('BASE TABLE', 'SYSTEM VERSIONED') order by table_name)
````

No solution on internet worked. This happens for DB_HOST set to 'localhost' or '127.0.0.1' . Driver is mysql. 
Database and user exists and I am able to connect to database using terminal. No tables created on my own.

I tried to composer dump-autoload and php artisan optimize:clear to no results.

This is similar issue to #19522 that is already closed but still not solved after all these years.

### Steps To Reproduce

1.Fresh install laravel
2. modify .env to reflect user and password and database
3. create database and user and make sure you can connect using terminal.
4. in project directory run `php artisan serve`
5. see the error

**Comment by mateusjunges** [comment]
Maybe [this](https://stackoverflow.com/questions/27365473/mamp-with-laravel-unix-socket) helps?

**Comment by Behinder** [comment]
If you say “maybe” it means you are not sure. I am not using MAMP and I am
not connecting using socket. And as I said clearing cache not helped.
It’s sad Laravel is shipped with something that not working out of the box.
This literally making me searching for alternatives, can you suggest one
beside Symfony? Language does not matter

W dniu niedz., 28.01.2024 o 07:23 Mateus Junges ***@***.***>
napisał(a):

**Comment by chrishow** [comment]
Does a raw PDO call succeed, or throw an exception?

eg:

```php
<?php

$pdo = new PDO(
	'mysql:host=localhost;dbname=<YOUR DB NAME>',
	'<YOUR MYSQL USERNAME>',
	'<YOUR MYSQL PASSWORD>'
);

var_dump($pdo);

```

**Comment by hamzaelmaghari** [comment]
The error message `SQLSTATE[HY000] [2002] No such file or directory` typically indicates that Laravel is unable to connect to the MySQL database. This could be due to a misconfiguration in your `.env` file, particularly the `DB_HOST` parameter.

Ensure that the `DB_HOST` parameter in your `.env` file is correctly set to the host where your MySQL server is running. If you're running MySQL locally, it's usually `127.0.0.1` or `localhost`. If your MySQL server is running on a different machine, you need to specify the IP address or domain name of that machine.

Once you've corrected the `DB_HOST` parameter, try running `php artisan migrate` again. If the issue persists, double-check your MySQL server's configuration and ensure it's running properly.
